src/ert/run_models/batch_simulator_run_model.py

```diff
@@ -0,0 +1,502 @@
+        )
+
+        # The SqliteStorage object is used to store optimization results from
+        # Seba in an sqlite database. It reacts directly to events emitted by
+        # Seba and is not called by Everest directly. The stored results are
+        # accessed by Everest via separate SebaSnapshot objects.
+        # This mechanism is outdated and not supported by the ropt package. It
+        # is retained for now via the seba_sqlite package.
+        seba_storage = SqliteStorage(  # type: ignore
+            optimizer, self.everest_config.optimization_output_dir
+        )
+
+        # Run the optimization:
+        optimizer_exit_code = optimizer.run().exit_code
+
+        # Extract the best result from the storage.
+        self._result = seba_storage.get_optimal_result()  # type: ignore
+
+        if self._monitor_thread is not None:
+            self._monitor_thread.stop()
+            self._monitor_thread.join()
+            self._monitor_thread = None
+
+        self._exit_code = (
+            "max_batch_num_reached"
+            if self._max_batch_num_reached
+            else optimizer_exit_code
+        )
+
+    def _handle_errors(
+        self,
+        batch: int,
+        simulation: Any,
+        realization: str,
+        fm_name: str,
+        error_path: str,
+    ) -> None:
+        fm_id = "b_{}_r_{}_s_{}_{}".format(batch, realization, simulation, fm_name)
+        logger = logging.getLogger("forward_models")
+        with open(error_path, "r", encoding="utf-8") as errors:
+            error_str = errors.read()
+
+        error_hash = hash(error_str)
+        err_msg = "Batch: {} Realization: {} Simulation: {} Job: {} Failed {}".format(
+            batch, realization, simulation, fm_name, "Error: {}\n {}"
+        )
+
+        if error_hash not in self._fm_errors:
+            error_id = len(self._fm_errors)
+            logger.error(err_msg.format(error_id, error_str))
+            self._fm_errors.update({error_hash: {"error_id": error_id, "ids": [fm_id]}})
+        elif fm_id not in self._fm_errors[error_hash]["ids"]:
+            self._fm_errors[error_hash]["ids"].append(fm_id)
+            error_id = self._fm_errors[error_hash]["error_id"]
+            logger.error(err_msg.format(error_id, ""))
+
+    def _simulation_callback(self, ctx: BatchContext | None) -> None:
+        logging.getLogger(EVEREST).debug("Simulation callback called")
+        if ctx is None:
+            return
+        if self._monitor_thread is not None:
+            self._monitor_thread.stop()
+
+        self._monitor_thread = _MonitorThread(
+            context=ctx,
+            error_callback=self._handle_errors,
+            callback=self._sim_callback,
+            delete_run_path=self._simulation_delete_run_path,
+            display_all_jobs=self._display_all_jobs,
+        )
+        self._monitor_thread.start()
+
+    def _ropt_callback(
+        self, _: Event, optimizer: OptimizationPlanRunner, simulator: Simulator
```

Comment:
This break in a next release, since that definition moved in `ropt`, so then we have to change the import. We can make our life a bit easier by not typing that here and removing the import. Maybe leave a TODO to type it again later. Or leave it and will fix it when it breaks...

Reply:
I think it should be fine to leave it typed now and just update when it changes, we expect mypy to be happy with this, and updating the import should be OK. We will get mypy failures on PRs / main as soon as ROPT updates